src/server.c

```diff
@@ -4795,6 +4806,22 @@ struct redisCommand *lookupCommandOrOriginal(robj **argv ,int argc) {
     return cmd;
 }
 
+static int shouldPropagate(int target) {
+    if (!server.replication_allowed || target == PROPAGATE_NONE || server.loading)
+        return 0;
+
+    if (target & PROPAGATE_AOF) {
+        if (server.aof_state != AOF_OFF)
+            return 1;
+    }
+    if (target & PROPAGATE_REPL) {
+        if (server.masterhost == NULL && (server.repl_backlog || listLength(server.slaves) != 0))
```

Comment:
Where did this come from and why did you have to add it?

Reply:
it came from `replicationFeedSlaves` and it's here (along with the AOF condition above) in order to prevent pushing stuff to also_propagate in case we don't have any aspect of replication (CPU + memory optimization)

do you think this condition should be in a function that replicationFeedSlaves uses? IMHO it's basic-enough to appear in those two places